libs/utils/src/celix_hash_map.c

```diff
@@ -237,11 +235,12 @@ static void celix_hashMap_addEntry(celix_hash_map_t* map, unsigned int hash, con
     newEntry->next = entry;
     map->buckets[bucketIndex] = newEntry;
     if (map->size++ >= celix_hashMap_threshold(map)) {
-        celix_hashMap_resize(map, 2 * map->bucketsSize);
+        return celix_hashMap_resize(map, 2 * map->bucketsSize);
```

Comment:
if `newEntry` is added but `celix_hashMap_resize` failed, should it return SUCCESS or error code?

Reply:
For weakly stored key, it will make huge difference.

Reply:
IMO it should fail, else we have a error which is not reported or logged to celix_err without the users knowing a issue occurred. 

Refactored the hashmap resize so that this is done before creating a new entry. When resize fails - or when creating a new entry fails - users still have ownership of weakly stored keys. I also updated the header dox for the storeKeysWeakly to clarify this.